### ISSUE TITLE
Parse Error in Javascript with Array Destructuring

### ISSUE THREAD

**Issue by brendongo** [semgrep/semgrep#580]
Fail to parse in cases when the first element of the destructing is ignored and the rest is assigned.

https://semgrep.live/Jpo

Putting the example here:

This fails
```
array = [1,2,3,4,5];
var [, ...x] = array
```
but 
```
array = [1,2,3,4,5];
var [a, ...x] = array
```
and 
```
array = [1,2]
var [,x] = array
```
parses correctly so it is not an issue with ignoring or the ... operator but something when they interact together.

**Environment**
Using version 0.4.9

**Comment by brendongo** [comment]
Exists in the wild here: https://github.com/lodash/lodash/blob/master/tail.js#L18

**Comment by aryx** [comment]
problably one of the TODO in the pattern() function in pfff/lang_js/analyze/ast_js_build.ml

**Comment by aryx** [comment]
To test the CST of JS you can use ~/pfff/pfff -dump_js, and for the AST -dump_ast_js, so you can quickly see which one fails.

**Comment by nbrahms** [comment]
Should be fixed in 0.10 (@brendongo reminder that we'll need to bump `pfff` prior to release).
